pkg/cluster/check/default.go

```diff
@@ -73,14 +79,6 @@ func DefaultClusterChecks() []ClusterCheck {
 				return K8sPodReadyAssertion(ctx, cluster, "kube-system", "k8s-app=kube-dns")
 			}, 3*time.Minute, 5*time.Second)
 		},
-	}
-}
-
-// ExtraClusterChecks returns a set of additional Talos cluster readiness checks which work only for newer versions of Talos.
-//
-// ExtraClusterChecks can't be used reliably in upgrade tests, as older versions might not pass the checks.
-func ExtraClusterChecks() []ClusterCheck {
-	return []ClusterCheck{
 		// wait for all the nodes to be schedulable
 		func(cluster ClusterInfo) conditions.Condition {
 			return conditions.PollingCondition("all k8s nodes to report schedulable", func(ctx context.Context) error {
```

Comment:
as we're no longer doing anything for 0.5, we can promote this check to be default